ts/src/bybit.ts

```diff
@@ -7498,117 +5668,121 @@ export default class bybit extends Exchange {
             }
             results.push (this.parsePosition (rawPosition, market));
         }
-        return this.filterByArrayPositions (results, 'symbol', symbols, false);
+        return this.filterByArray (results, 'symbol', symbols, false);
     }
 
-    async fetchDerivativesPositions (symbols: string[] = undefined, params = {}) {
-        await this.loadMarkets ();
-        const request = {};
-        let market = undefined;
-        let settle = undefined;
+    async fetchPositions (symbols: string[] = undefined, params = {}) {
+        /**
+         * @method
+         * @name bybit#fetchPositions
+         * @description fetch all open positions
+         * @see https://bybit-exchange.github.io/docs/v5/position
+         * @param {string[]|undefined} symbols list of unified market symbols
+         * @param {object} [params] extra parameters specific to the bybit api endpoint
+         * @returns {object[]} a list of [position structure]{@link https://docs.ccxt.com/#/?id=position-structure}
+         */
+        let symbol = undefined;
         if (Array.isArray (symbols)) {
             const symbolsLength = symbols.length;
             if (symbolsLength > 1) {
                 throw new ArgumentsRequired (this.id + ' fetchPositions() does not accept an array with more than one symbol');
+            } else if (symbolsLength === 1) {
+                symbol = symbols[0];
             }
-            if (symbolsLength === 1) {
-                market = this.market (symbols[0]);
-                settle = market['settle'];
-                request['symbol'] = market['id'];
+        } else if (symbols !== undefined) {
+            symbol = symbols;
+        }
+        await this.loadMarkets ();
+        const [ enableUnifiedMargin, enableUnifiedAccount ] = await this.isUnifiedEnabled ();
+        const isUnifiedAccount = (enableUnifiedMargin || enableUnifiedAccount);
+        const request = {};
+        let market = undefined;
+        let type = this.safeString (params, 'type', 'linear');
```

Comment:
@sc0Vu this is not the way to check the type, if the symbol is available we should use that, otherwise we should use `handleMarketTypeAndParams` and `handleSubTypeAndParams`